src/storage/btree.c

```diff
@@ -14350,12 +14351,32 @@ btree_find_boundary_leaf (THREAD_ENTRY * thread_p, BTID * btid, VPID * pg_vpid,
   /* read the root page */
   P_vpid.volid = btid->vfid.volid;
   P_vpid.pageid = btid->root_pageid;
-  P_page = pgbuf_fix (thread_p, &P_vpid, OLD_PAGE, PGBUF_LATCH_READ, PGBUF_UNCONDITIONAL_LATCH);
-  if (P_page == NULL)
+
+  do
```

Comment:
My guess this pattern, where we try to fix the page, wait of replication, and retry, will be required in many places.
I suggest adding a new wrapper function which does this (returns the page latched or with other error than ER_PAGE_AHEAD_OF_REPLICATION)
This function could be in page_buffer if it can be reused by heap module.

Reply:
should this be extracted to a different issue? What do you say @cristiarg?

Reply:
Yes, it could be extracted in a function in page_buffer, near the existing `pgbuf_check_page_ahead_of_replication` wher e tdes `page_desync_lsa` is set and reused: here, in `heap_get_visible_version_with_repl_desync`.
Also in that new function, add some logging with same LSA as from the message:
```
1275 Page %1d|%2d is ahead of replication. Page LSA is %3lld|%4d, replication LSA is %5lld|%6d.
```
such that it is obvious where page desync is encountered and where it is handled. Some `const char*` context argument will be needed to provide meaningful contextualized message.